### PR TITLE
Implement github actions ci script

### PR DESCRIPTION
**Pull request by SchleichsSalaticus**
Moving saphanabootstrap-formula CI  from Travis to GitHub Actions
The CI automatically publishes new releases to the Open Build Service project every time a pull request is merged into master.

### CODE REVIEW

File: .github/workflows/formulaCI.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+name: Formula CI
+# Description: "Moving saphanabootstrap-formula from Travis to Github Actions"
```
Comment (arbulu89) [inline .github/workflows/formulaCI.yml:2]: I would change this description by something like this: https://github.com/SUSE/habootstrap-formula/blob/master/.github/workflows/formula-ci.yml#L3

File: .github/workflows/formulaCI.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Add salt repository
+      uses: myci-actions/add-deb-repo@4
+      with:
+        repo: deb http://repo.saltstack.com/apt/ubuntu/18.04/amd64/2019.2/ bionic main
```
Comment (arbulu89) [inline .github/workflows/formulaCI.yml:20]: Can we take advantage here and use the latest salt version?
Here an example: https://github.com/SUSE/habootstrap-formula/blob/master/.github/workflows/formula-ci.yml#L23

File: .github/workflows/formulaCI.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+jobs:
+
+  validation:
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
```
Comment (arbulu89) [inline .github/workflows/formulaCI.yml:16]: I would remove this empty line, it makes more difficult to understand the `yml`

File: .github/workflows/formulaCI.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    - name: Add salt repository
+      uses: myci-actions/add-deb-repo@4
+      with:
+        repo: deb http://repo.saltstack.com/apt/ubuntu/18.04/amd64/2019.2/ bionic main
+        key-server: http://repo.saltstack.com/apt/ubuntu/18.04/amd64/2019.2/SALTSTACK-GPG-KEY.pub
+
```
Comment (arbulu89) [inline .github/workflows/formulaCI.yml:22]: Remove these empty lines

File: .github/workflows/formulaCI.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    needs: validation
+    runs-on: ubuntu-18.04
+    if: ${{ github.event_name != 'pull_request' }}  
+    container: 
+      image: shap/continuous_deliver
+      env:
+        OBS_USER: ${{ secrets.OBS_USER }}
+        OBS_PASS: ${{ secrets.OBS_PASS }}
+        OBS_PROJECT: ${{ secrets.OBS_PROJECT }}
+
```
Comment (arbulu89) [inline .github/workflows/formulaCI.yml:40]: Remove empty line

File: .github/workflows/formulaCI.yml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    steps:
+    - uses: actions/checkout@v2 
+      with:
+        fetch-depth: 0
+# OSC credentials must be configured beforehand as the HOME variables cannot be changed from /github/home
```
Comment (arbulu89) [inline .github/workflows/formulaCI.yml:45]: Better to indent these comment to the correct level, to `- name` level.

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Hi @SchleichsSalaticus ,
I have added some comment inline. Besides them:
1. Remove the `.travis.yml` file. We don't use this CI anymore, so better to have the house clean
2. It would be nice that we added the badge of the CI state in the README file. Here an example (the url will be different of course): https://github.com/SUSE/habootstrap-formula/commit/4126486dca767766b3b82147b3c0f64140f7c566

File: .github/workflows/formulaCI.yml (hunk 1 above)
Comment (SchleichsSalaticus) [reply .github/workflows/formulaCI.yml:2]: I add the comment description like in the example

File: .github/workflows/formulaCI.yml (hunk 2 above)
Comment (SchleichsSalaticus) [reply .github/workflows/formulaCI.yml:20]: Yes we can  :D 
I have updated the static version to the latest 
        repo: deb http://repo.saltstack.com/apt/ubuntu/18.04/amd64/latest/ bionic main
        key-server: http://repo.saltstack.com/apt/ubuntu/18.04/amd64/latest/SALTSTACK-GPG-KEY.pub

File: .github/workflows/formulaCI.yml (hunk 6 above)
Comment (SchleichsSalaticus) [reply .github/workflows/formulaCI.yml]: Hi @arbulu89 did you mean like this ? 
-name: 
# comment 
# comment

File: .github/workflows/formulaCI.yml [hunk 7]
```diff
@@ -0,0 +1,78 @@
+name: Formula CI
+# this workflow will
+# - run formula validation step - test/validate-formula.sh
```
Comment (arbulu89) [inline .github/workflows/formulaCI.yml:3]: Just as a comment, this is wrong. It should be `ci/validate-formula.sh`

File: .github/workflows/formulaCI.yml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  validation:
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Add salt repository
+      uses: myci-actions/add-deb-repo@4
+      with:
+        repo: deb http://repo.saltstack.com/apt/ubuntu/18.04/amd64/latest/ bionic main
```
Comment (arbulu89) [inline .github/workflows/formulaCI.yml:20]: Please, don't use `latest`. Latest might bring some not tested elements. In openSUSE we ship `salt v3000`, so let's use this

**Review by arbulu89 — APPROVED** [review]
@SchleichsSalaticus Looks good!
Check my 2 comments:
- Use salt 3000 version
- Fix comment

File: .github/workflows/formulaCI.yml (hunk 7 above)
Comment (SchleichsSalaticus) [reply .github/workflows/formulaCI.yml]: i updated the salt version from latest to 3000 and corrected the comment

**Comment by SchleichsSalaticus** [comment]
All required changes have been made :D

**Pull request merged**